### PR TITLE
#162727463 Email Notification API

### PR DESCRIPTION
**Pull request by wombolo**
#### What does this PR do?
Ensure Email with Subject and message body is delivered to specified recipient Email address

#### Description of Task to be completed?
Deliver Emails from the application via ElasticEmail 3rd party API to specified recipient

#### How should this be manually tested?
-Import ./src/helpers/EmailNotificationAPI.js into the intended file for use.

-Then paste, in order to instantiate the EmailNotificationAPI class
`const sampleEmail = new EmailNotificationAPI({
  recipient: "didun.morodolu@andela.com",
  subject: "Subject Line",
  message: "<h3> Hello, World from Merry world </h3>"
});
`

`sampleEmail.sendEmail()
  .then(res => console.log(res))
  .catch(err => (err.message));
`
or use async - await method

#### Any background context you want to provide?
The class requires recipient email, a subject line and message which can be in HTML format when instantiating

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162727463

### CODE REVIEW

**Review by daniellamarr — APPROVED** [review]

File: src/helpers/EmailNotificationAPI.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import nodemailer from 'nodemailer';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
```
Comment (julietezekwe) [inline src/helpers/EmailNotificationAPI.js:5]: JSdoc is required for commenting. 
See sample here [Js doc](http://usejsdoc.org/)
Nice work though

File: src/helpers/EmailNotificationAPI.js [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 
 dotenv.config();
 
 class EmailNotificationAPI {
 
   constructor(emailPayload){
     this.mailOptions = {
-      from: '"Merry Ah 👻" <team@merry-ah.io>', // sender address
+      from: `"Merry Ah 👻" <${process.env.EMAILUSER}>`, // sender address
       to: emailPayload.recipient, // list of receivers
```
Comment (Proception) [inline src/helpers/EmailNotificationAPI.js:17]: Does this allow for push to many users?

File: src/helpers/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import EmailNotificationAPI from 'EmailNotificationAPI';
```
[the remark this answers is not in the document]
Comment (andela-dbamidele) [reply src/helpers/index.js]: export {default as EmailNotificationAPI} from './'EmailNotificationAPI';

[the remark this answers is not in the document]
Comment (andela-dbamidele) [reply src/helpers/index.js]: even better `export * from './'EmailNotificationAPI'`

File: src/helpers/EmailNotificationAPI.js (hunk 2 above)
Comment (wombolo) [reply src/helpers/EmailNotificationAPI.js:17]: Not yet, Feature will come later on

File: src/helpers/EmailNotificationAPI.js (hunk 1 above)
Comment (wombolo) [reply src/helpers/EmailNotificationAPI.js:5]: JSdocs have been added

File: src/helpers/EmailNotificationAPI.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+   */
+  async sendEmail() {
+    const { mailOptions } = this.mailOptions;
+
+    try {
+      const mail = await EmailNotificationAPI.transportCreator()
+        .sendMail(mailOptions);
+      if (mail.response.includes('OK')) return ('Message sent');
+    } catch (error) {
+      return (error);
```
Comment (andela-dbamidele) [inline src/helpers/EmailNotificationAPI.js]: Why do you have to put the return value in parentheses?

**Review by andela-dbamidele — CHANGES_REQUESTED** [review]

File: src/helpers/EmailNotificationAPI.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+   * @Represents sendEmail
+   * @return {string|Object} Success message | Error
+   */
+  async sendEmail() {
+    const { mailOptions } = this.mailOptions;
+
+    try {
+      const mail = await EmailNotificationAPI.transportCreator()
+        .sendMail(mailOptions);
+      if (mail.response.includes('OK')) return ('Message sent');
```
Comment (andela-dbamidele) [inline src/helpers/EmailNotificationAPI.js]: You didn't put this in curly braces too.

File: src/helpers/EmailNotificationAPI.js (hunk 4 above)
Comment (wombolo) [reply src/helpers/EmailNotificationAPI.js]: I intended returning more than 1 variables before, but its no longer needed now.

**Pull request merged**